hsmd/libhsmd.c

```diff
@@ -2297,7 +2297,7 @@ u8 *hsmd_init(struct secret hsm_secret,
 	 * incompatibility detection) with alternate implementations.
 	 */
 	return take(towire_hsmd_init_reply_v4(
-			    NULL, 4,
+			    NULL, 5,
```

Comment:
I don't think this should be hard-coded.  I think it should be negotiated in `init_hsm` (in `hsmd.c`) and passed into here:
```
/* use the highest protocol version that both sides support */
mutual_version = min(maxversion, 5); /* maxversion is from lightningd, 5 is for the hsmd code */
```

Reply:
I increase this version because this check was failing https://github.com/ElementsProject/lightning/blob/897ed22f67c76a3e4b2f2adfcaf92b101898339d/lightningd/hsm_control.c#L145

idk if we do need to change the check-in there for the min version supported by the deamon?

Reply:
1. `hsm_control.c` sends `hsmd_init` with min=5 and max=5
2. `init_hsm` in `hsmd.c` should determine the highest protocol level that is mutually supported 
3. this version is then returned in the `hsmd_init_reply_v4 message

Reply:
thanks, this logic should be implemented inside https://github.com/ElementsProject/lightning/pull/7088/commits/2a66de88aa405fb7b792dc2f7734430a922a6c68